internal/gocodegen/marshalling.go

```diff
@@ -0,0 +1,473 @@
+			List(Id("v"), Err()).Op(":=").Qual("encore.dev/types/uuid", "FromString").Call(Id("s")),
+			Id("e").Dot("setErr").Call(Lit("invalid parameter"), Id("field"), Err()),
+			Return(Id("v")),
+		}}
+	case schema.Builtin_TIME:
+		fn = methodDescription{true, "ToTime", String(), Qual("time", "Time"), false, []Code{
+			List(Id("v"), Err()).Op(":=").Qual("time", "Parse").Call(Qual("time", "RFC3339"), Id("s")),
+			Id("e").Dot("setErr").Call(Lit("invalid parameter"), Id("field"), Err()),
+			Return(Id("v")),
+		}}
+	case schema.Builtin_USER_ID:
+		fn = methodDescription{true, "ToUserID", String(), Qual("encore.dev/beta/auth", "UID"), false, []Code{
+			Return(Qual("encore.dev/beta/auth", "UID").Call(Id("s"))),
+		}}
+	case schema.Builtin_JSON:
+		fn = methodDescription{true, "ToJSON", String(), Qual("encoding/json", "RawMessage"), false, []Code{
+			Return(Qual("encoding/json", "RawMessage").Call(Id("s"))),
+		}}
+	default:
+		type kind int
+		const (
+			unsigned kind = iota + 1
+			signed
+			float
+		)
+		numTypes := map[schema.Builtin]struct {
+			typ  string
+			kind kind
+			bits int
+		}{
+			schema.Builtin_INT8:    {"int8", signed, 8},
+			schema.Builtin_INT16:   {"int16", signed, 16},
+			schema.Builtin_INT32:   {"int32", signed, 32},
+			schema.Builtin_INT64:   {"int64", signed, 64},
+			schema.Builtin_INT:     {"int", signed, 64},
+			schema.Builtin_UINT8:   {"uint8", unsigned, 8},
+			schema.Builtin_UINT16:  {"uint16", unsigned, 16},
+			schema.Builtin_UINT32:  {"uint32", unsigned, 32},
+			schema.Builtin_UINT64:  {"uint64", unsigned, 64},
+			schema.Builtin_UINT:    {"uint", unsigned, 64},
+			schema.Builtin_FLOAT64: {"float64", float, 64},
+			schema.Builtin_FLOAT32: {"float32", float, 32},
+		}
+
+		def, ok := numTypes[t]
+		if !ok {
+			return "", errors.Newf("unsupported type: %s", t)
+		}
+
+		cast := def.typ != "int64" && def.typ != "uint64" && def.typ != "float64"
+		var err error
+		fn = methodDescription{true, "To" + strings.Title(def.typ), String(), Id(def.typ), false, []Code{
+			List(Id("x"), Err()).Op(":=").Do(func(s *Statement) {
+				switch def.kind {
+				case unsigned:
+					s.Qual("strconv", "ParseUint").Call(Id("s"), Lit(10), Lit(def.bits))
+				case signed:
+					s.Qual("strconv", "ParseInt").Call(Id("s"), Lit(10), Lit(def.bits))
+				case float:
+					s.Qual("strconv", "ParseFloat").Call(Id("s"), Lit(def.bits))
+				default:
+					err = errors.Newf("unknown kind %v", def.kind)
+				}
+			}),
+			Id("e").Dot("setErr").Call(Lit("invalid parameter"), Id("field"), Err()),
+			ReturnFunc(func(g *Group) {
+				if cast {
+					g.Id(def.typ).Call(Id("x"))
+				} else {
+					g.Id("x")
+				}
+			}),
+		}}
+		if err != nil {
+			return "", err
+		}
+	}
+
+	b.seenBuiltins[key] = fn
+	b.builtins = append(b.builtins, fn)
+	return fn.Method, nil
+}
+
+func (b *MarshallingCodeGenerator) builtinToString(t schema.Builtin, slice bool) (string, error) {
```

Comment:
This method is brand new and does the reverse of the above - but I based it by copy/pasting the above method and then tweaking